### PR TITLE
Add connector validation to otelcol

### PR DESCRIPTION
**Pull request by djaglowski**
Part of #6700 

This PR finishes adding connectors to the otelcol config including validation. Validation for connectors is the same as with other components, except two additional checks are included:
1. That each connector is used as an exporter in at least one pipeline, and also as a receiver in at least one pipeline.
2. That each connector ID is distinct from receivers and exporters. (Depends on decision on https://github.com/open-telemetry/opentelemetry-collector/pull/6927#issuecomment-1400791808)

### CODE REVIEW

File: otelcol/config.go [hunk 1]
```diff
@@ -101,6 +101,20 @@ func (cfg *Config) Validate() error {
...
 	}
 
+	// Validate the connector configuration.
+	for connID, connCfg := range cfg.Connectors {
+		if err := component.ValidateConfig(connCfg); err != nil {
+			return fmt.Errorf("connectors::%s: %w", connID, err)
+		}
+
+		if _, ok := cfg.Exporters[connID]; ok {
+			return fmt.Errorf("connectors::%s: cannot have same id as exporter", connID)
```
Comment (jpkrohling) [inline otelcol/config.go:111]: Putting myself in the shoes of someone who's looking at this message for the first time, I think the version below might be easier to grasp what's going on.

```suggestion
			return fmt.Errorf("connectors::%s: there's already an exporter named %q", connID, connID)
```

**Review by jpkrohling — COMMENTED** [review]
LGTM, just perhaps one error message could be improved.

File: otelcol/config_test.go [hunk 2]
```diff
@@ -59,6 +62,14 @@ func (nc *nopProcConfig) Validate() error {
 	return nc.validateErr
 }
 
+type nopConnConfig struct {
+	validateErr error
+}
+
+func (nc *nopConnConfig) Validate() error {
+	return nc.validateErr
+}
```
Comment (bogdandrutu) [inline otelcol/config_test.go:71]: Separate PR, please merge all these config structs. We merged the config some time ago, so they don't need to be a struct per component type.

**Review by bogdandrutu — APPROVED** [review]

File: otelcol/otelcoltest/config_test.go [hunk 3]
```diff
@@ -70,6 +71,7 @@ func TestLoadConfigAndValidate(t *testing.T) {
 	factories, err := NopFactories()
 	assert.NoError(t, err)
 
+	require.NoError(t, featuregate.GlobalRegistry().Apply(map[string]bool{"otelcol.enableConnectors": true}))
```
Comment (bogdandrutu) [inline otelcol/otelcoltest/config_test.go:74]: Should you `defer` to the previous state?

**Review by bogdandrutu — APPROVED** [review]

File: otelcol/otelcoltest/config_test.go (hunk 3 above)
Comment (djaglowski) [reply otelcol/otelcoltest/config_test.go:74]: Done.

File: otelcol/config_test.go (hunk 2 above)
Comment (djaglowski) [reply otelcol/config_test.go:71]: Ok, I will work on this next.

**Pull request merged**